Automattic/pocket-casts-android
# Support edge-to-edge for the filters section

## Description

This change focuses on converting all the screens in the filters tab to edge-to-edge for the Android 15 project. Please ignore the other tabs for now. I fixed some pages in other tabs, but I will focus on these sections later.

There will be multiple pull requests required to support Android 15 edge-to-edge but I'm going to break them down into smaller PRs. I will merge them all into a `task/android-15` branch, and after all the work is completed, raise a PR to merge into `main`.

## Testing Instructions

Please try tapping around in the filters tab to find issues with the edge-to-edge changes. Areas I have discovered issues with are:
- Older versions of Android versions
- Themes
- Content under the system bars 
- End of pages / lists under the bottom navigation or mini player

Here are some test steps I have followed in case you want to try them too:

1. Open the filters tab
2. ✅ Verify the next pages aren't under the status bar and you can get to the bottom of the page, even with the mini player open.
3. Tap on a filter
4. Tap on the three dots -> Filter settings
5. Go back and create a filter
6. Try the different filter options such as "Episode Status" and "Release Date"

## Screenshots

Android 15, SDK 35

| Before | After |
| --- | --- |
| ![filters-before-list](https://github.com/user-attachments/assets/94c37c1c-cf98-473d-b9ec-9f4942e45c64) | ![filters-after-list](https://github.com/user-attachments/assets/9bcfc845-ece6-44d3-aa4b-f1c181967876) |
| ![filters-before-list-dark](https://github.com/user-attachments/assets/d9761097-0dfe-4fc7-a19c-2dd19147c1b2) | ![filters-after-list-dark](https://github.com/user-attachments/assets/ec1d015f-9b94-45f8-bf51-234aa3c4b93f) |
| ![filters-before-list-rose](https://github.com/user-attachments/assets/58c0a333-fe68-46ed-8787-37ea56bd359e) | ![filters-after-list-rose](https://github.com/user-attachments/assets/f4508dfa-436b-405d-8e1d-1b2af9e00851) |
| ![filters-before-list-classic](https://github.com/user-attachments/assets/7a48863b-1fbc-436c-b52b-0671dbcd1655) | ![filters-after-list-classic](https://github.com/user-attachments/assets/ae1d8146-e93e-4724-a4ba-57335910b99e) |
| ![filters-before-list-radiation](https://github.com/user-attachments/assets/8ca2e150-5f91-4b1e-b698-c25c970b201d) | ![filters-after-list-radiation](https://github.com/user-attachments/assets/69f28d27-c587-43ca-9674-733e90904a24) |
| ![filters-before-empty](https://github.com/user-attachments/assets/1f45bfd7-e35d-4a77-a17c-008c45e0dcad) | ![filters-after-empty](https://github.com/user-attachments/assets/77645884-0ed6-466f-900c-9595ee3678f6) |
| ![filters-before-settings](https://github.com/user-attachments/assets/972bf6e6-9bd9-4042-83c1-f3717bbd5250) | ![filters-after-settings](https://github.com/user-attachments/assets/ecd1e8cc-59ca-4584-a7d6-2cacde2ea326) |
| ![filters-before-create](https://github.com/user-attachments/assets/2713a3de-d7f9-4a9b-82a3-73d186dd49da) | ![filters-after-create](https://github.com/user-attachments/assets/7411b8e3-94d3-4d82-ad96-b3981e477d54) |
| ![filters-before-choose-podcasts](https://github.com/user-attachments/assets/a7318b35-9a80-4ad6-a33e-c40241b5f80a) | ![filters-after-choose-podcasts](https://github.com/user-attachments/assets/dce44e6f-fe48-4242-9736-379bbd316494) |
| ![filters-before-episode-status](https://github.com/user-attachments/assets/f55398e7-07fe-4e36-8e08-6c017b54e430) | ![filters-after-episode-status](https://github.com/user-attachments/assets/7af03089-3872-4b86-948a-99bd3a19920d) |
| ![filters-before-release-date](https://github.com/user-attachments/assets/6b96b7d4-5aa7-4fa3-98c7-00bb0a9af02d) | ![filters-after-release-date](https://github.com/user-attachments/assets/72c18416-2a5f-4a6d-bf81-83c23c984498) |
| ![filters-before-episode-duration](https://github.com/user-attachments/assets/3c52caab-a684-4ef8-aa8f-39411a3a74fb) | ![filters-after-episode-duration](https://github.com/user-attachments/assets/2558a50a-4004-452a-8bef-7fd853dafcfc) |
| ![filters-before-download](https://github.com/user-attachments/assets/ba2aef79-3197-463f-b4f0-a8b21f86aa07) | ![filters-after-download](https://github.com/user-attachments/assets/e31a58b5-6db4-4b58-a4ee-7c90a8f1cadc) |
| ![filters-before-media-type](https://github.com/user-attachments/assets/73b5b354-5a9e-47d2-a547-0f1c2d94d639) | ![filters-after-media-type](https://github.com/user-attachments/assets/a241be5d-fa6e-4b7b-b434-e8d591a83c9a) |
| ![filters-before-list-landscape](https://github.com/user-attachments/assets/054419db-b846-4d2e-99aa-963c5f2cba48) | ![filters-after-list-landscape](https://github.com/user-attachments/assets/cfd50855-7c4f-48d7-bb0e-fc3606d14354) |

Android 11, SDK 30
| Before | After |
| --- | --- |
| ![filters-before](https://github.com/user-attachments/assets/d3a185e5-be6e-4dbb-96b9-5dc9f4a442ea) | ![filters-after](https://github.com/user-attachments/assets/3962844f-0bde-4b89-a017-d9d64e95527a) |
| ![filters-create-before](https://github.com/user-attachments/assets/c3848bc7-5dfd-4a45-bf71-fc2736a2170f) | ![filters-create-after](https://github.com/user-attachments/assets/31445ad3-d084-4d8c-89e1-77c2d109bc58) |
| ![filters-episode-status-before](https://github.com/user-attachments/assets/87941925-a031-4850-8453-13414ae8652c) | ![filters-episode-status-after](https://github.com/user-attachments/assets/492277b8-4510-4b4f-9978-3faedea7004c) |
| ![filters-dark-before](https://github.com/user-attachments/assets/d4c8ea47-a6ca-4045-b17d-ca99b5486c29) | ![filters-dark-after](https://github.com/user-attachments/assets/9cf88695-6fab-461d-a75c-453ce7d01a07) |
| ![filters-create-dark-before](https://github.com/user-attachments/assets/1407579c-dfab-497b-9bb5-57efdd3dd1df) | ![filters-create-dark-after](https://github.com/user-attachments/assets/dab2ddd2-df7b-47d5-9827-99049f5b0918) |
| ![filters-episode-status-dark-before](https://github.com/user-attachments/assets/1b999e31-530e-434a-b0e4-fc33ae5a9703) | ![filters-episode-status-dark-after](https://github.com/user-attachments/assets/123d365c-3af4-43fc-8922-6cfd41fd6cfa) |
| ![filters-rose-before](https://github.com/user-attachments/assets/ba3d4364-5a9a-459a-ab38-cf64a4571af7) | ![filters-rose-after](https://github.com/user-attachments/assets/3bab01fd-84e3-4e81-925a-4f6d61aea3bc) |
| ![filters-create-rose-before](https://github.com/user-attachments/assets/c6acd20b-b22a-4162-b604-fcfc511f1053) | ![filters-create-rose-after](https://github.com/user-attachments/assets/4c9c4067-6f5b-4761-a923-972e7c0d47ee) |
| ![filters-episode-status-rose-before](https://github.com/user-attachments/assets/2eba6788-39bf-4bba-ae47-4e5385b848e3) | ![filters-episode-status-rose-after](https://github.com/user-attachments/assets/d897eef3-6a9f-4931-9190-1464a4552e07) |



## Checklist
 
#### I have tested any UI changes...
